rerun_py/rerun_sdk/rerun2/components/point2d_ext.py

```diff
@@ -0,0 +1,19 @@
+from __future__ import annotations
+
+__all__ = ["Point2DArrayExt"]
+
+from typing import Any, Sequence, Type
+
+import numpy as np
+import pyarrow as pa
+
+
+class Point2DArrayExt:
+    @staticmethod
+    def _from_similar(data: Any | None, *, mono: type, mono_aliases: Type, many: type, many_aliases: Type, arrow: type):
+        if isinstance(data, Sequence) and (len(data) > 0 and isinstance(data[0], mono)):
+            arrays = [np.asarray(datum) for datum in data]
+        else:
+            arrays = np.require(np.asarray(data), np.float32).reshape((-1, 2)).tolist()
```

Comment:
tolist might be quite expensive here. Can't we go from np to arrow more directly like in the other places?

Reply:
We very likely can, I didnt really try to optimize in any way; I'll see if i can quickly whip up something nicer